### PR TITLE
[Merged by Bors] - PLAT-615: rewrite integration test for message "from the future"

### PR DESCRIPTION
**Pull request by aalekseev-ru**
**- What I did**
rewritee predicate for futureSM

### CODE REVIEW

File: ledger-core/conveyor/sm_future_event.go [hunk 1]
```diff
@@ -19,30 +19,30 @@ import (
 // Before such stop, this SM will attempt to capture and fire a termination handler for the event.
 
 func newFutureEventSM(pn pulse.Number, ps *PulseSlot, createFn smachine.CreateFunc) smachine.StateMachine {
-	return &futureEventSM{wrapEventSM{pn: pn, ps: ps, createFn: createFn}}
+	return &FutureEventSM{wrapEventSM{pn: pn, ps: ps, createFn: createFn}}
```
Comment (cyraxred) [inline ledger-core/conveyor/sm_future_event.go:22]: please keep it private

File: ledger-core/testutils/journal/journal.go [hunk 2]
```diff
@@ -102,10 +103,10 @@ func (p *Journal) WaitStopOf(sample smachine.StateMachine, stopCount int) syncki
 		).AfterPositiveToZero)
 }
 
-func (p *Journal) WaitAnyActivityOf(sample smachine.StateMachine, stopCount int) synckit.SignalChannel {
+func (p *Journal) WaitFutureSlotInited(stopCount int) synckit.SignalChannel {
 	return p.WaitOnce(
 		predicate.NewCounter(
-			predicate.NewSMTypeFilter(sample, nil),
+			predicate.NewSMTypeFilter(&conveyor.FutureEventSM{}, predicate.BeforeStep((&conveyor.FutureEventSM{}).StepWaitMigration)),
```
Comment (cyraxred) [inline ledger-core/testutils/journal/journal.go:109]: as mentioned earlier - you are mixing different cases

and Journal is NOT a right place for this specific predicate as well

File: ledger-core/conveyor/sm_future_event.plantuml [hunk 3]
```diff
@@ -1,30 +1,30 @@
 @startuml
 state "GetInitStateFor" as T00_S001
-T00_S001 : futureEventSM
+T00_S001 : FutureEventSM
```
Comment (bigbes) [inline ledger-core/conveyor/sm_future_event.plantuml:3]: please regen plantuml

Comment (aalekseev-ru) [reply ledger-core/conveyor/sm_future_event.plantuml]: done

**Comment by aalekseev-ru** [comment]
bors merge

File: ledger-core/virtual/integration/method_test.go [hunk 4]
```diff
@@ -1096,35 +1098,94 @@ func Test_CallMethodWithBadIsolationFlags(t *testing.T) {
...
+		Arguments:           insolar.MustSerialize([]interface{}{}),
 	}
 
-	server.IncrementPulse(ctx)
+	// now we are not legitimate sender
+	jetCoordinatorMock.QueryRoleMock.Return([]reference.Global{server.RandomGlobalWithPulse()}, nil)
+	jetCoordinatorMock.MeMock.Return(server.RandomGlobalWithPulse())
 
-	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
+	server.SendPayloadAsFuture(ctx, &pl)
```
Comment (cyraxred) [inline ledger-core/virtual/integration/method_test.go:1178]: I strongly recommend to add WaitIdleConveyor() before sending anything

Comment (aalekseev-ru) [reply ledger-core/virtual/integration/method_test.go:1179]: done

**Comment by aalekseev-ru** [comment]
bors merge

**Pull request closed without merging**